diff --git a/components/audio_hal/include/audio_hal.h b/components/audio_hal/include/audio_hal.h
index 1a9cf12a..065218c1 100755
--- a/data/build/homeassist03/components/audio_hal/include/audio_hal.h
+++ b/data/build/homeassist03/components/audio_hal/include/audio_hal.h
@@ -147,7 +147,7 @@ typedef struct audio_hal {
     esp_err_t (*audio_codec_set_volume)(int volume);                                                         /*!< set codec volume */
     esp_err_t (*audio_codec_get_volume)(int *volume);                                                        /*!< get codec volume */
     esp_err_t (*audio_codec_enable_pa) (bool enable);                                                        /*!< enable pa */
-    xSemaphoreHandle audio_hal_lock;                                                                         /*!< semaphore of codec */
+    SemaphoreHandle_t audio_hal_lock;                                                                         /*!< semaphore of codec */
     void *handle;                                                                                            /*!< handle of audio codec */
 } audio_hal_func_t;
 
diff --git a/components/esp_peripherals/include/esp_peripherals.h b/components/esp_peripherals/include/esp_peripherals.h
index ad4f9fe9..c6be4089 100644
--- a/data/build/homeassist03/components/esp_peripherals/include/esp_peripherals.h
+++ b/data/build/homeassist03/components/esp_peripherals/include/esp_peripherals.h
@@ -29,6 +29,7 @@
 #include "audio_event_iface.h"
 #include "audio_common.h"
 #include "freertos/event_groups.h"
+#include "freertos/timers.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -76,7 +77,7 @@ typedef struct esp_periph *esp_periph_handle_t;
 typedef esp_err_t (*esp_periph_func)(esp_periph_handle_t periph);
 typedef esp_err_t (*esp_periph_run_func)(esp_periph_handle_t periph, audio_event_iface_msg_t *msg);
 typedef esp_err_t (*esp_periph_event_handle_t)(audio_event_iface_msg_t *event, void *context);
-typedef void (*timer_callback)(xTimerHandle tmr);
+typedef void (*timer_callback)(TimerHandle_t tmr);
 
 /**
  * @brief Common peripherals configurations

 
